cpp/src/arrow/dataset/discovery.cc

```diff
@@ -210,10 +210,22 @@ Result<std::shared_ptr<DatasetFactory>> FileSystemDatasetFactory::Make(
 Result<std::shared_ptr<DatasetFactory>> FileSystemDatasetFactory::Make(
     std::string uri, std::shared_ptr<FileFormat> format,
     FileSystemFactoryOptions options) {
+  // TODO Partitioning support. Dictionary support should be done before that. See
+  // ARROW-12481.
   std::string internal_path;
   ARROW_ASSIGN_OR_RAISE(std::shared_ptr<fs::FileSystem> filesystem,
-                        arrow::fs::FileSystemFromUri(uri, &internal_path))
-  ARROW_ASSIGN_OR_RAISE(fs::FileInfo file_info, filesystem->GetFileInfo(internal_path))
+                        fs::FileSystemFromUri(uri, &internal_path))
+  ARROW_ASSIGN_OR_RAISE(arrow::fs::FileInfo file_info,
+                        filesystem->GetFileInfo(internal_path));
```

Comment:
nit: it seems we sometimes use `arrow::fs` and sometimes use `fs::`. Is there any special reason for that?

Reply:
Not special reason for that. `fs::` looks more reasonable since it's used from other places in this file. Changed to `fs::`.

Reply:
I see. Thanks.